lib/doorkeeper-mongodb/mixins/mongoid/application_mixin.rb

```diff
@@ -56,8 +57,26 @@ def by_uid_and_secret(uid, secret)
           def by_uid(uid)
             where(uid: uid.to_s).first
           end
+
+          def supports_confidentiality?
+            fields.include?("confidential")
+          end
         end
 
+        # Fallback to existing, default behaviour of assuming all apps to be
+        # confidential if the migration hasn't been run
+        def confidential
+          return super if self.class.supports_confidentiality?
+
+          ActiveSupport::Deprecation.warn 'You are susceptible to security bug ' \
+            'CVE-2018-1000211. Please follow instructions outlined in ' \
+            'Doorkeeper::CVE_2018_1000211_WARNING'
+
+          true
+        end
+
+        alias_method :confidential?, :confidential
```

Comment:
Technically, `confidential`, `supports_confidentiality?`, and this alias are all unnecessary if using Mongoid.  The field definition adds `confidential` and `confidential?` both, so if the field exists, all of this is redundant.  If the field doesn't exist, it is a little different, but in this branch, if you have an `Application`, you have the field, so it is pretty redundant.

Adding these are mostly for new ORMs (if they forget the field somehow?), or if this concern is used in a custom `Ajpplication` model (if that is even possible), a forked repository, and finally for as close of a compatibility with the main ActiveRecord code as is possible.